src/validation.cpp

```diff
@@ -1622,9 +1619,8 @@ static bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockInd
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
-                return state.DoS(100, error("ConnectBlock(): inputs missing/spent"),
-                                 REJECT_INVALID, "bad-txns-inputs-missingorspent");
+            if (!Consensus::CheckTxInputs(tx, state, view, pindex->nHeight, nFees))
```

Comment:
In commit "Optimization: Minimize the number of times"

If I understand correctly, it would be more correct to pass `SpendHeight(view)` instead of `pindex->nHeight` as the nSpendHeight argument, but in this case it doesn't matter because the value is only used for coinbase transactions, and this isn't a coinbase transaction? Or maybe there are other reasons why `pindex->nHeight` is ok to pass?

Either way, I think this is confusing, and that there should be a comment here explaining the `pindex->nHeight` value. Or maybe you could delete the `nSpendHeight` argument and have `CheckTxInputs` call `GetSpendHeight(inputs)` when needed. This seems like it would be a simplification and I don't think there should be a performance cost because `CheckTxInputs` seems to be skipped for coinbase transactions in all cases except for one call that invokes `GetSpendHeight` anyway.